core/src/nucleus/mod.rs

```diff
@@ -49,17 +54,31 @@ impl ZomeFnCall {
             // @see https://github.com/holochain/holochain-rust/issues/198
             id: snowflake::ProcessUniqueId::new(),
             zome_name: zome.to_string(),
-            cap_name: capability.to_string(),
+            cap: cap,
             fn_name: function.to_string(),
             parameters: parameters.into(),
         }
     }
 
     pub fn same_fn_as(&self, fn_call: &ZomeFnCall) -> bool {
         self.zome_name == fn_call.zome_name
-            && self.cap_name == fn_call.cap_name
+            && self.cap == fn_call.cap
             && self.fn_name == fn_call.fn_name
     }
+
+    pub fn cap_name(&self) -> String {
+        match self.cap.clone() {
+            Some(call) => call.cap_name,
+            None => panic!("null cap call unimplemented!"),
+        }
+    }
+
+    pub fn cap_token(&self) -> Address {
+        match self.cap.clone() {
+            Some(call) => call.cap_token,
+            None => panic!("null cap call unimplemented!"),
```

Comment:
Wondering if these getters should return an Option and panic should happen in the context that uses `ZomeFnCall`?

Reply:
All the places that use that are going to change and disappear, so though I agree theoritically, I don't think I want to bother to fix it right now.  Will do so in next PR.